intro-to-R-tidyverse/04a-intro_to_R_exercise.Rmd

```diff
@@ -0,0 +1,191 @@
+that we performed a set of [differential expression analyses on](./scripts/00-setup-intro-to-R.R).
+
+### Set Up 
+
+Use this chunk to load the `tidyverse` library. 
+
+```{r}
+
+```
+
+## Read in the gene results file
+
+Use `readr::read_tsv` to read in the file "gene_results_GSE44971.tsv" and assign 
+it the name `stats_df`. 
+
+```{r}
+stats_df <-
+```
+
+Use this chunk to explore what your data frame, `stats_df` looks like. 
+
+```{r}
+
+```
+
+## Read in the metadata
+
+Use `readr::read_tsv` to read in the file "cleaned_metadata_GSE44971.tsv" and assign 
+it the name `metadata`. 
+
+```{r}
+metadata <- 
+```
+
+Use this chunk to explore what your data frame, `metadata` looks like. 
+
+```{r}
+
+```
+
+### Subsetting variables
+
+Use subsetting to take a look at the `avg_expression` column. 
+Hint: `$`
+
+```{r}
+
+```
+
+Use the `min()` argument to find what the minimum average expression in this dataset is. 
+Remember you can use `?min` or the help panel to find out more about a function. 
+
+```{r}
+# Find the minimum average expression value
+
+```
+
+Find the `log`, using base 2, of the average expression values. 
+
+```{r}
+# Find the log of base 2 of the average expression
+
+```
+
+## Using logical arguments
+
+Use subsetting to take a look at the `adj_p_value` column. 
+
+```{r}
+
+```
+
+Find out which p-values in total are below a `0.05` cutoff using a logical 
+statement. 
+
+```{r}
+
+```
+
+Name the logical vector you created above as `significant_vector`.
+
+```{r}
+
+```
+
+Use `sum` to count how many p values in the total set are below this cutoff. 
+
+```{r}
+
+```
+
+## Filter the dataset
+
+Subset the column `contrasts` from `stats_df`. 
+
+```{r}
+
+```
+
+Construct a logical vector around the contrasts variable you subset above that 
+indicates which rows of data in `stats_df` are from the `astrocytoma_normal` 
+contrast test. 
+
+```{r}
+
+```
+
+Use `dplyr::filter` to keep only the data for the `astrocytoma_normal` contrast
+in `stats_df`. 
+
+```{r}
+astrocytoma_normal_df <- dplyr::filter(<DATA_FRAME>, <LOGICAL_VECTOR>)
+```
+
+Use the `nrow` argument on `astrocytoma_normal_df` to see if your filter worked. 
+You should have `2268` rows. 
+
+```{r}
+nrow(astrocytoma_normal_df)
+```
+
+Do the same filtering as you did above, but this time, use a `%>%` by removing 
+`stats_df` from the first argument position and piping it into the filter. 
+
+```{r}
+astrocytoma_normal_df <- <DATA_FRAME> %>% 
+  dplyr::filter(<LOGICAL_VECTOR>)
+```
+
+Save your filtered data to a TSV file using `readr::write_tsv`. 
+Call it `astrocytoma_normal_contrast_results.tsv` and save it to the `results`
+directory. 
+
+```{r}
+
+```
+
+### Create a density plot 
+
+Set up a ggplot object for `astrocytoma_normal_df` and set `x` as the average 
+expression variable. 
+Use the `+` to add on a layer called `geom_density()`
+
+```{r}
+ggplot(<DATA_FRAME>, aes(x = <VARIABLE_NAME>)) + 
```

Comment:
Is there a reason why you are explicitly calling functions from their respective package libraries above (i.e. `readr::write_tsv`) but not here for the `ggplot2` functions?

Reply:
I debated about removing the `readr::` bit but I think in some instances its good for referencing purposes so the participants know where these functions come from. Perhaps we'll add a note about what package its from and reference it with `::` in the text, but without such in the actual code? The inconsistency needs to be fixed, but will have to figure out what the best way is to fix it.

Reply:
> Perhaps we'll add a note about what package its from and reference it with :: in the text, but without such in the actual code?

I believe this a good idea but we can probably use a 2nd (and probably 3rd) opinion on this.
We have them load in the `tidyverse` library at the beginning of the notebook so I would opt for having the `readr::` in text but make use of the functionality that loading in a library allows you (not having to use the `readr::` in the code).

Reply:
I think its okay too that we use `::` in the solutions, so that they know where the functions are coming from. We don't want to force them to either the `library` or `::` strategy, so I think that was my original thought on this, was to have both in the solutions. Perhaps I could make a note explaining that the `::` aren't necessary but are there for informational purposes.

Reply:
Ahh okay, I understand. I liked what you did in the 4b exercise notebook, allowing them to choose between loading in the tidyverse library or going on to use the `::` throughout the notebook. 

Is that something you did not want to implement in this first exercise notebook for informational purposes?

Reply:
First notebook that will be one extra thing for them to think about. Trying to ease them into the process.